pkg/parser/parser.go

```diff
@@ -358,3 +387,278 @@ func (p *parser[T]) SetFilters(filters []string) error {
 	p.filterSpecs = filterSpecs
 	return nil
 }
+
+// Prometheus related stuff
+
+func (p *parser[T]) AttrsGetter(colNames []string) (func(any) []attribute.KeyValue, error) {
+	columnMap := p.columns.GetColumnMap()
+
+	keys := []attribute.Key{}
+	getters := []func(*T) attribute.Value{}
+
+	for _, colName := range colNames {
+		col, ok := columnMap.GetColumn(colName)
+		if !ok {
+			return nil, fmt.Errorf("unknown column: %s", colName)
+		}
+
+		var getter func(*T) attribute.Value
+
+		switch col.Kind() {
+		case reflect.Int:
+			getter = func(a *T) attribute.Value {
+				ff := columns.GetFieldFunc[int, T](col)
```

Comment:
`ff` can be cached by moving it outside of the getter. See also below.

Reply:
(Not relevant anymore as used `GetFieldAsNumberFunc` as suggested by you)